### DISCUSSION TITLE
Disable tree view while refreshing?

### DISCUSSION THREAD

**Discussion by gee4vee** [microsoft/vscode-discussions#2123] [Extension Development QnA] [answered]
Is it possible to disable the nodes in the tree view (or the contextual menu items) while the tree is refreshing? I've seen in our custom extension that if a user tries to invoke a context menu item while the tree is refreshing, it causes undefined property errors and other issues.

**Comment by burkeholland** [comment]
Can you provide a simple reproduceable example here? I think this might be possible using context keys and updating them during the refresh process. For instance....

```
"contributes": {
    "menus": {
      "view/item/context": [
        {
          "command": "yourExtension.yourCommand",
          "when": "view == yourTreeViewId && !treeRefreshing"
        }
      ]
    }
  }
 ```

**Reply by gee4vee** [reply]
That makes sense but that would remove the item if the keys don’t match. Is there a way to disable context menu items? I’ve seen that happen in the source control view container tree.

**Reply by isidorn** [reply]
Yes, you can also disable via context menus. I think if your command is disabled, and the menu is still there this should work.
@alexr00 might be able to help. Or you should try asking GH Copilot.

**Reply by gee4vee** [reply]
I could not find anything in the api docs about how to disable context menu items without removing them. Are there any examples out there? I could not find anything.

**Reply by gjsjohnmurray** [reply, marked as answer]
> I could not find anything in the api docs about how to disable context menu items without removing them. Are there any examples out there? I could not find anything.

You can set a when clause on the `enablement` property of the command in the `contributes.commands` section of your package.json
